sdk/core/azure_core_amqp/src/cbs.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corp. All Rights Reserved.
+// cspell: words amqp sasl
+
+use azure_core::error::Result;
+use std::fmt::Debug;
+
+use super::session::AmqpSession;
+
+#[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+type CbsImplementation = super::fe2o3::cbs::Fe2o3ClaimsBasedSecurity;
+
+#[cfg(any(not(any(feature = "enable-fe2o3-amqp")), target_arch = "wasm32"))]
+type CbsImplementation = super::noop::NoopAmqpClaimsBasedSecurity;
+
+pub trait AmqpClaimsBasedSecurityTrait {
+    /// Asynchronously attaches the Claims-Based Security (CBS) node to the AMQP session.
+    ///
+    /// This method is responsible for setting up the necessary AMQP links for CBS operations.
+    /// It must be called before attempting to authorize any AMQP paths using the `authorize_path` method.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` which is:
+    /// - `Ok(())` on successful attachment of the CBS node.
+    /// - `Err(e)` where `e` is an error from the `azure_core::error::Result` indicating the failure reason.
+    ///
+    fn attach(&self) -> impl std::future::Future<Output = Result<()>> {
+        async { unimplemented!() }
+    }
```

Comment:
Async traits are supported now, and our MSRV of 1.76 asserts that:
```suggestion
    async fn attach(&self) -> Result<()>;
```

Reply:
Async traits do not appear to be supported:

```
  --> sdk\core\azure_core_amqp\src\connection.rs:32:5
   |
32 |     async fn open(
   |     ^^^^^
   |
   = note: you can suppress this lint if you plan to use the trait only in your own code, or do not care about auto traits like `Send` on the `Future`
   = note: `#[warn(async_fn_in_trait)]` on by default
help: you can alternatively desugar to a normal `fn` that returns `impl Future` and add any desired bounds such as `Send`, but these cannot be relaxed without a breaking API change
   |
32 ~     fn open(
33 |         &self,
 ...
36 |         options: Option<AmqpConnectionOptions>,
37 ~     ) -> impl std::future::Future<Output = Result<()>> + Send {async {
38 |         unimplemented!();
39 ~     } }
   |
```

Reply:
Ah, yes: <https://blog.rust-lang.org/2023/12/28/Rust-1.75.0.html#async-fn-and-return-position-impl-trait-in-traits>

Perhaps we should do its second suggestion for now:

> you can alternatively desugar to a normal `fn` that returns `impl Future` and add any desired bounds such as `Send`, but these cannot be relaxed without a breaking API change

Curious what @yoshuawuyts thinks. We do declare our `rust-version` as 1.76 currently, so I don't think this is a real issue. A bit surprised the lint isn't taking that into consideration, though maybe it only does that for the current crate and not scan the entire dependency tree (`azure_core` declares the MSRV currently, though I'll move that to `typespec`).

Reply:
I'm pretty sure that the "desugar to a normal fn that returns impl Future" is what I'm currently doing.

If it's not what I'm doing, please let me know how I misinterpreted the suggestion.